source/language/standard_library.rst

```diff
@@ -0,0 +1,329 @@
+
+Implementations should consider the ``stdgates.inc`` file to be locatable, regardless of any other
+settings that they might apply around include search paths.
+
+Upon interpreting an ``include "stdgates.inc";`` statement, implementations should make available
+exactly the set of gates defined in the standard library for the language version that matches the
+:ref:`version-string statement <version-string>`, if present.  If the version statement is not
+present, the implementation should make available the gates from the language version it is parsing
+the rest of the program under.
+
+Implementations must not define gates that are not present in chosen OpenQASM 3 version, where
+doing so could cause a user program to be ill formed (for example due to a user attempting to define
+their own gate with a clashing name).
+
+An implementation is not required to supply a literal ``stdgates.inc`` file to the user.
+
+An implementation must make all the gates available as if they were defined with ``gate``
+statements with contents that match the mathematical descriptions given here, but it is explicitly
+permissible for implementations to internally treat the ``stdgates.inc`` include as a series of
+``defcal`` statements for all qubits used in the program.
+
+
+API Documentation
+-----------------
+
+The content of the standard library is documented below, including the OpenQASM version that each
+gate was introduced in.
+
+Single-qubit gates
+..................
+
+.. gate:: p(λ) a
+
+   The phase gate.  Defined by the mapping :math:`\lvert0\rangle \to \lvert0\rangle` and
+   :math:`\lvert1\rangle \to e^{i\lambda}\lvert1\rangle`.
+
+   Equivalent to ``ctrl @ gphase(λ) a``.
+
+   .. seealso::
+
+      :gate:`rz`
+         The same operation up to a global phase.
+
+      :gate:`phase`
+         An alternative name for backwards compatibility with OpenQASM 2.
+
+   .. versionadded:: 3.0
+
+.. gate:: x a
+
+   The Pauli X gate.  Defined by the mapping :math:`\lvert0\rangle \leftrightarrow \lvert1\rangle`.
+
+   .. versionadded:: 3.0
+
+.. gate:: y a
+
+   The Pauli Y gate.  Defined by the mapping :math:`\lvert0\rangle \to i\lvert1\rangle` and
+   :math:`\lvert1\rangle \to -i\lvert0\rangle`.
+
+   .. versionadded:: 3.0
+
+.. gate:: z a
+
+   The Pauli Z gate.  Defined by the mapping :math:`\lvert0\rangle \to \lvert0\rangle` and
+   :math:`\lvert1\rangle \to \lvert1\rangle`.
+
+   Equivalent to ``p(pi) a``.
+
+   .. versionadded:: 3.0
+
+.. gate:: h a
+
+   The Hadamard gate.  Defined by the mapping
+   :math:`\lvert0\rangle \to \bigl(\lvert0\rangle + \lvert1\rangle\bigr)/\sqrt2` and
+   :math:`\lvert1\rangle \to \bigl(\lvert0\rangle - \lvert1\rangle\bigr)/\sqrt2`.
+
+   .. versionadded:: 3.0
+
+.. gate:: s a
+
+   The :math:`\sqrt Z` gate (see :gate:`z`).  The square root is chosen conventionally, that is
+   the gate is equivalent to :math:`P(\pi/2)`, in terms of :gate:`p`.
```

Comment:
I had deliberately used the mathematical gate symbols, then linked them to the associated OpenQASM gate, to try and make them understandable at a glance to people familiar with the maths, but I've no problem accepting changes on it if people don't think it's clear.

Reply:
I don't mind too much, but I did find myself wondering what `P` was and whether it was supposed to be the gate `p`.

Reply:
Maybe my real question is where do I find the mathematical definition of `P`? Is there a standard definition?

Reply:
To avoid all doubt, I've pushed 42fd895 which includes formal mathematics and symbols for each gate.  I've deliberately avoided using matrix representations, and I was very explicit about the tensor indices of each qubit in each example.

Reply:
Thanks @jakelishman. The formal basis transformation definitions are perhaps overkill but definitely cool and resolves the uncertainty I was worried about.